cub/cub/device/device_reduce.cuh

```diff
@@ -593,7 +594,7 @@ public:
                                           _CUDA_EXEC::determinism::run_to_run_t>;
 
     // Query relevant properties from the environment
-    auto stream = _CUDA_STD_EXEC::__query_or(env, ::cuda::get_stream, ::cuda::stream_ref{});
+    auto stream = _CUDA_STD_EXEC::__query_or(env, ::cuda::get_stream, ::cuda::stream_ref{static_cast<cudaStream_t>(0)});
```

Comment:
Q: Is this the prefered way to construct the `cudaStream_t`? IIUC, it's a pointer, so `cudaStream_t{nullptr}` or even `cudaStream_t{}` would work. Casting an integral zero to a pointer type is a bit of a type violation to me, but I know that's what `NULL` does and it's legal C++.